### PR TITLE
ExplodeyBat (SCP-2398)

### PR DESCRIPTION
**Pull request by 77seven77seven**
## About the Pull Request

we need new content and this good yes (its in LCZ)
How 2398 Works:
Step 1. Acquire funny bat
Step 2. Swing funny bat at human
Step 3. Small explosion happens, ur target gets gibbed
Step 4. you get damaged a little from the explosion but if you have some helpers you will be alive
Step 5. ???
Step 6. Profit

## Why It's Good For The Game

We need new content and this is cool, I guess.

## Changelog

:cl:
add: Added the explodey bat (SCP-2398)
add: Made the storage room in LCZ a little smaller to accommodate the SCP-2398 Containment Chamber
/:cl:

### CODE REVIEW

**Comment by aidan-bowman** [comment]
Don't make it gib, the user, just the victim. Explosion should also be dependent on victims size if possible. To balance it out, swinging the bat should require a do_after (like in real life, you need to get the proper stance to actually swing it). If you want to make it even funnier, the bat could have some interactions when thrown too.

**Comment by aidan-bowman** [comment]
[Link to SCP article for easy access](https://scp-wiki.wikidot.com/scp-2398)

**Comment by aidan-bowman** [comment]
Also, it's missing documentation in the archives.

**Comment by 77seven77seven** [comment]
> Don't make it gib, the user, just the victim. Explosion should also be dependent on victims size if possible. To balance it out, swinging the bat should require a do_after (like in real life, you need to get the proper stance to actually swing it). If you want to make it even funnier, the bat could have some interactions when thrown too.

That will just make it an overpowered SCP item and people will aggro grab people, swing it and make their target unrevivable. And that will just be so fucking annoying to deal with, especially with Mauricio.

**Comment by aidan-bowman** [comment]
> > Don't make it gib, the user, just the victim. Explosion should also be dependent on victims size if possible. To balance it out, swinging the bat should require a do_after (like in real life, you need to get the proper stance to actually swing it). If you want to make it even funnier, the bat could have some interactions when thrown too.
> 
> That will just make it an overpowered SCP item and people will aggro grab people, swing it and make their target unrevivable. And that will just be so fucking annoying to deal with, especially with Mauricio.

If explosion is size dependant, you could probably make it so swinging the bat at a human causes pretty severe damage to the user. If it just gibs them, it's probably going to go ignored by most players.

File: code/modules/SCP/SCPs/SCP-2398.dm
Comment (MACIEKBAKI) [inline code/modules/SCP/SCPs/SCP-2398.dm]: alternatively, make the SCP indestructible?

Comment (MACIEKBAKI) [reply code/modules/SCP/SCPs/SCP-2398.dm]: could work

File: code/modules/SCP/SCPs/SCP-2398.dm
Comment (MACIEKBAKI) [inline code/modules/SCP/SCPs/SCP-2398.dm]: var names could be more descriptive

File: code/modules/SCP/SCPs/SCP-2398.dm
Comment (MACIEKBAKI) [inline code/modules/SCP/SCPs/SCP-2398.dm]: also, a way to balance this such that it's not useless should be probably considered

**Comment by TheDarkElites** [comment]
If I may, a good idea might be making the gibbing of the attacker be a percentage chance rather than guaranteed. It should also be noted that this isn't a PVP game but rather a game about the SCPs themselves, and therefore critically changing an SCP for the purpose of PVP balance is somewhat silly. You would have to steel the bat, and if you did and intended to use it against others, site security could just shoot you.

**Comment by aidan-bowman** [comment]
> If I may, a good idea might be making the gibbing of the attacker be a percentage chance rather than guaranteed. It should also be noted that this isn't a PVP game but rather a game about the SCPs themselves, and therefore critically changing an SCP for the purpose of PVP balance is somewhat silly. You would have to steel the bat, and if you did and intended to use it against others, site security could just shoot you.

Exactly. Using ANY SCP to powergame is a complete violation of the rules, and I'd assume that it would be hard to gain access to this item for the people who might actually be able to abuse it (e.g. traitors). If the SD grabs the bat and starts massacring class D, that's not an issue with the bat, that's an issue with the admins.

**Comment by 77seven77seven** [comment]
> > If I may, a good idea might be making the gibbing of the attacker be a percentage chance rather than guaranteed. It should also be noted that this isn't a PVP game but rather a game about the SCPs themselves, and therefore critically changing an SCP for the purpose of PVP balance is somewhat silly. You would have to steel the bat, and if you did and intended to use it against others, site security could just shoot you.
> 
> Exactly. Using ANY SCP to powergame is a complete violation of the rules, and I'd assume that it would be hard to gain access to this item for the people who might actually be able to abuse it (e.g. traitors). If the SD grabs the bat and starts massacring class D, that's not an issue with the bat, that's an issue with the admins.

*cough* _MAURICIO_ *cough*

**Comment by TheDarkElites** [comment]
> _cough MAURICIO cough_

Yes, but balancing it this way means when testing (one of the intended things in the game), you will always kill the D-Class. Now there is some leniency for this if it's part of the SCP, but as this is an attempt at balancing, you end up making D-Class even more disposable, which is hardly interesting for them.
Also, as Pizza said, we must trust in good faith that we can rely on admins once we are up and running again. 
You could include a warning sent to admins in case SCP-2398 is being repeatedly used.
Also, a minor nitpick, you should probably put the actual SCP into the PR title or at least the colloquial term for clairty.

**Comment by 77seven77seven** [comment]
> > _cough MAURICIO cough_
> 
> Yes, but balancing it this way means when testing (one of the intended things in the game), you will always kill the D-Class. Now there is some leniency for this if it's part of the SCP, but as this is an attempt at balancing, you end up making D-Class even more disposable, which is hardly interesting for them. Also, as Pizza said, we must trust in good faith that we can rely on admins once we are up and running again. You could include a warning sent to admins in case SCP-2398 is being repeatedly used. Also, a minor nitpick, you should probably put the actual SCP into the PR title or at least the colloquial term for clairty.

the name is a reference to my previous PR 'BurnGuy'

**Comment by TheDarkElites** [comment]
I'm aware, but I still feel for clarity's sake, the name should be more specific.

File: code/modules/SCP/SCPs/SCP-2398.dm
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-2398.dm]: done

File: code/modules/SCP/SCPs/SCP-2398.dm
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-2398.dm]: this is now invalidated

File: code/modules/SCP/SCPs/SCP-2398.dm
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-2398.dm]: they could be, but its literally stated what they are

**Comment by aidan-bowman** [comment]
> > _cough MAURICIO cough_
> 
> Yes, but balancing it this way means when testing (one of the intended things in the game), you will always kill the D-Class. Now there is some leniency for this if it's part of the SCP, but as this is an attempt at balancing, you end up making D-Class even more disposable, which is hardly interesting for them.

Oh I forgot about testing. Personally I wouldn't worry about testing with skips too much until we actually develop a research system. Besides, with the way the tests in the article are written, it'd probably be more hitting everything EXCEPT actual humans.

**Comment by TheDarkElites** [comment]
> Personally I wouldn't worry about testing with skips too much until we actually develop a research system.

Once I'm done with my 173 rework, I'm developing a research system, I've already got a system figured out, and I'm gonna present it to head devs and maintainers once my last 173 PR is merged, so research should be at least somewhat considered.

**Comment by Tsurupeta** [comment]
I want user to not be gibbed, but instead his arm fractured when he hits victim. 

Also it would be funny if when thrown, it could hit people, gib them and search in radius of screen for other person to bounce into. Although it would feel real bad on recieving side.

**Comment by 77seven77seven** [comment]
> I want user to not be gibbed, but instead his arm fractured when he hits victim.
> 
> Also it would be funny if when thrown, it could hit people, gib them and search in radius of screen for other person to bounce into. Although it would feel real bad on recieving side.

it would be funny if i knew how to do that

**Comment by aidan-bowman** [comment]
> It would be funny if when thrown, it could hit people, gib them and search in radius of screen for other person to bounce into. Although it would feel real bad on recieving side.

Just have it bounce randomly, no need for heat seeking explosive baseball. Also, thrown object can be caught by also being in throw mode.

**Comment by 77seven77seven** [comment]
> > It would be funny if when thrown, it could hit people, gib them and search in radius of screen for other person to bounce into. Although it would feel real bad on recieving side.
> 
> Just have it bounce randomly, no need for heat seeking explosive baseball. Also, thrown object can be caught by also being in throw mode.

that's the thing, i don't know how to do that

**Comment by 77seven77seven** [comment]
review please :)

**Comment by 77seven77seven** [comment]
> I want user to not be gibbed, but instead his arm fractured when he hits victim.
> 
> Also it would be funny if when thrown, it could hit people, gib them and search in radius of screen for other person to bounce into. Although it would feel real bad on recieving side.

also the person swinging now is injured :)

**Review by MACIEKBAKI — APPROVED** [review]
code ok i guess; did not review map changes

**Comment by 77seven77seven** [comment]
> code ok i guess; did not review map changes

image
map changes

**Comment by aidan-bowman** [comment]
> > code ok i guess; did not review map changes
> 
> image map changes

You didn't put the documentation in the archives?

**Comment by 77seven77seven** [comment]
> > > code ok i guess; did not review map changes
> > 
> > 
> > image map changes
> 
> You didn't put the documentation in the archives?

I mean, who uses archives, and... Is it really worth the trouble? Even 457 doesn't have documentation in the archives.

**Comment by aidan-bowman** [comment]
> > > > code ok i guess; did not review map changes
> > > 
> > > 
> > > image map changes
> > 
> > 
> > You didn't put the documentation in the archives?
> 
> I mean, who uses archives, and... Is it really worth the trouble? Even 457 doesn't have documentation in the archives.

average maintainer (spawn of evil) "i need to make sure the codebase is clean and healthy!!!"
vs average shitcoder (agent of heaven) "Someone else will fix it later."

**Comment by TheDarkElites** [comment]
> I mean, who uses archives, and... Is it really worth the trouble?

Yes because the only reason archives are not in use is because they have not yet been fleshed out. You cant just not do something because its not in use atm, because when that system is designed and your pr isint made for it, it becomes alot more of a pain in the ass to fix. I.E you know you're adding an SCP, so adding it to the archives takes like a minute, but someone else would have to go through the archives first and then find what SCPs are missing. So yes the "trouble" (like two minutes of your time) is worth it. Also its funny you brought up 457 because you _also_ coded it, you cant refer to your own actions as justification for a future mistake. So yeah you should probably add archive documentation and make a PR to add some for 457.

**Comment by 77seven77seven** [comment]
> > I mean, who uses archives, and... Is it really worth the trouble?
> 
> Yes because the only reason archives are not in use is because they have not yet been fleshed out. You cant just not do something because its not in use atm, because when that system is designed and your pr isint made for it, it becomes alot more of a pain in the ass to fix. I.E you know you're adding an SCP, so adding it to the archives takes like a minute, but someone else would have to go through the archives first and then find what SCPs are missing. So yes the "trouble" (like two minutes of your time) is worth it. Also its funny you brought up 457 because you _also_ coded it, you cant refer to your own actions as justification for a future mistake. So yeah you should probably add archive documentation and make a PR to add some for 457.

Okay, it probably adds itself automatically since I can't see anything special in StrongDMM.

**Comment by TheDarkElites** [comment]
> Okay, it probably adds itself automatically since I can't see anything special in StrongDMM.

Dont assume, ask tsurupeta to be sure. If it was automatic I dont think cheese would be asking about you adding it in.

**Comment by 77seven77seven** [comment]
> > Okay, it probably adds itself automatically since I can't see anything special in StrongDMM.
> 
> Dont assume, ask tsurupeta to be sure. If it was automatic I dont think cheese would be asking about you adding it in.

It would be really fucking easy to add the papers in automatically even for my experience, why would some seasoned veteran of coding use the mapping way?

**Comment by TheDarkElites** [comment]
> It would be really fucking easy to add the papers in automatically even for my experience, why would some seasoned veteran of coding use the mapping way?

Ask just to be sure? Not like it would kill you. Here Ill ask for you- 
image

Also just got some weird dejavu

**Comment by 77seven77seven** [comment]
> > It would be really fucking easy to add the papers in automatically even for my experience, why would some seasoned veteran of coding use the mapping way?
> 
> Ask just to be sure? Not like it would kill you. Here Ill ask for you- image
> 
> Also just got some weird dejavu

I'm going to kill myself

**Comment by aidan-bowman** [comment]
> > I mean, who uses archives, and... Is it really worth the trouble?
> 
> Yes because the only reason archives are not in use is because they have not yet been fleshed out. You cant just not do something because its not in use atm, because when that system is designed and your pr isint made for it, it becomes alot more of a pain in the ass to fix. I.E you know you're adding an SCP, so adding it to the archives takes like a minute, but someone else would have to go through the archives first and then find what SCPs are missing. So yes the "trouble" (like two minutes of your time) is worth it. Also its funny you brought up 457 because you _also_ coded it, you cant refer to your own actions as justification for a future mistake. So yeah you should probably add archive documentation and make a PR to add some for 457.

To add onto this, if something doesn't look well-made or integrated with the game, the best thing to do is to try and improve it, not cause it to fall further behind the codebase.

**Comment by 77seven77seven** [comment]
2398 document has been added to archives, merg pls :)

**Comment by aidan-bowman** [comment]
> 2398 document has been added to archives, merg pls :)

Can you make it work on a broader set of mobs? Both mob/living/carbon and mob/living/simple_animal should be hittable.

Also, make explosion dependant on mob_size.

**Comment by 77seven77seven** [comment]
> > 2398 document has been added to archives, merg pls :)
> 
> Can you make it work on a broader set of mobs? Both mob/living/carbon and mob/living/simple_animal should be hittable.
> 
> Also, make explosion dependant on mob_size.

guess i can try

**Comment by 77seven77seven** [comment]
> > 2398 document has been added to archives, merg pls :)
> 
> Can you make it work on a broader set of mobs? Both mob/living/carbon and mob/living/simple_animal should be hittable.
> 
> Also, make explosion dependant on mob_size.

making the explosion dependent on mob_size will kill the user because even the light impact range is kind of crazy, and for some reason simple_mobs wouldn't just trigger it...
that's it, just carbons right now

**Comment by aidan-bowman** [comment]
> > > 2398 document has been added to archives, merg pls :)
> > 
> > 
> > Can you make it work on a broader set of mobs? Both mob/living/carbon and mob/living/simple_animal should be hittable.
> > Also, make explosion dependant on mob_size.
> 
> making the explosion dependent on mob_size will kill the user because even the light impact range is kind of crazy, and for some reason simple_mobs wouldn't just trigger it... that's it, just carbons right now

that's why you spend 2 hours carefully engineering a complex mathematical function that returns overbalanced values for the 5 mob sizes we have

**Comment by 77seven77seven** [comment]
> > > > 2398 document has been added to archives, merg pls :)
> > > 
> > > 
> > > Can you make it work on a broader set of mobs? Both mob/living/carbon and mob/living/simple_animal should be hittable.
> > > Also, make explosion dependant on mob_size.
> > 
> > 
> > making the explosion dependent on mob_size will kill the user because even the light impact range is kind of crazy, and for some reason simple_mobs wouldn't just trigger it... that's it, just carbons right now
> 
> that's why you spend 2 hours carefully engineering a complex mathematical function that returns overbalanced values for the 5 mob sizes we have

:(

**Comment by 77seven77seven** [comment]
pls merge :)

File: code/modules/SCP/SCPs/SCP-2398.dm
Comment (MACIEKBAKI) [inline code/modules/SCP/SCPs/SCP-2398.dm]:
```suggestion
	desc = "A generic wooden bat. The letters 'K.O.' are branded into the wood, just above the handle."
```

File: code/modules/SCP/SCPs/SCP-2398.dm
Comment (MACIEKBAKI) [inline code/modules/SCP/SCPs/SCP-2398.dm]: im sure theres a proc to format this for admins already; go find it

**Review by MACIEKBAKI — CHANGES_REQUESTED** [review]

File: code/modules/SCP/SCPs/SCP-2398.dm
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-2398.dm]: god fucking DAMN IT

Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-2398.dm]: 'i THINK  theres a proc for this... go find it lmao'

Comment (TheDarkElites) [reply code/modules/SCP/SCPs/SCP-2398.dm]: yeah thats kind of what all of us gotta do when we need to find the right way to do something :/

Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-2398.dm]: ;(

Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-2398.dm]: fixed

**Comment by 77seven77seven** [comment]
okay so far all the changes have been accomodated

**Comment by 77seven77seven** [comment]
The documents have been put into the archive, so far. I don't see any problems with it. Review if you find any, but so far.
codors, pls merg

File: code/modules/SCP/SCPs/SCP-2398.dm
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-2398.dm]: committed, it's now clearer

**Comment by TheDarkElites** [comment]
Might be a cool idea to have some sort of damage inflicted on the user if the target of the bat is an object or scp. Also maybe just have the user have their arm broken(I think this was suggested already). Uh I think you gotta look at the med code to see how to do that but I assume its probably a proc.

**Comment by 77seven77seven** [comment]
> Might be a cool idea to have some sort of damage inflicted on the user if the target of the bat is an object or scp. Also maybe just have the user have their arm broken(I think this was suggested already). Uh I think you gotta look at the med code to see how to do that but I assume its probably a proc.

ill get right on that

**Comment by 77seven77seven** [comment]
> Might be a cool idea to have some sort of damage inflicted on the user if the target of the bat is an object or scp. Also maybe just have the user have their arm broken(I think this was suggested already). Uh I think you gotta look at the med code to see how to do that but I assume its probably a proc.

the hitting SCPs and objects mechanic will be added after the new research mechanics will be added so it can be actually useful and not just some fluff which will make coding more of a hell for me, so far it breaks arms tho

**Comment by 77seven77seven** [comment]
other than that this is ready for merge i guess

**Comment by 77seven77seven** [comment]
merg ples

**Review by MACIEKBAKI — APPROVED** [review]

**Pull request merged**
